core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -473,7 +479,14 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
     }
 
     def commit() {
-      consumer.commitSync(offsets.map { case (tp, offset) =>  (tp, new OffsetAndMetadata(offset, ""))}.asJava)
+      val existingTopics: Set[String] = Try(consumer.listTopics) match {
```

Comment:
Should we do this in case of Timeout error and clear offsets map? Checking for every commit maybe costly.

Reply:
@omkreddy  Thanks for the response. Currently, `commitSync` will retry indefinitely when committing offsets for a nonexistent topic, and user could do nothing to get rid of this. That's why I always check the existence of topics before `commitSync` is called. Does it make any sense?

Reply:
With KIP-266, even `commitSync` will eventually timeout (see the use of `default.api.timeout.ms`). I prefer @omkreddy's suggestion since listing all the topics in the cluster can be expensive.

Reply:
With the `trunk` code, `commitSync` indefinitely retried instead of throwing `TimeoutException` when the topic was deleted. Here were the steps I reproduced:
1. Create a test topic named `test`
2. Write a simple consumer to commit offsets for this topic every 3 seconds using `commitSync`
3. Delete this topic
Later, the consumer complained "Offset commit failed on partition test-0 at offset...." forever. Did I miss anything?